### PR TITLE
Only opening connections when access token is provided

### PR DESCRIPTION
**Pull request by aasimkhan30**
Doing this when access token (like sql auth) is not provided leads to connection errors.

### CODE REVIEW

File: src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs [hunk 1]
```diff
@@ -38,10 +38,16 @@ public static async Task<TreeNode[]> Expand(string connectionString, SecurityTok
         {
             using (SqlConnection conn = new SqlConnection(connectionString))
             {
-                conn.AccessToken = accessToken?.Token;
-                conn.Open();
+                if(accessToken != null)
+                {
+                    conn.AccessToken = accessToken.Token;
+                    await conn.OpenAsync();
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs:44]: Shouldn't you still be trying to open even if there isn't an access token? Ideally this code should be generic enough that it can be used for other scenarios - such as ones where there's a user id/password.

Comment (aasimkhan30) [reply src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs:44]: Ideally yes, but unfortunately opening the sql auth based connections leads to connection exception in this line.
https://github.com/microsoft/sqltoolsservice/blob/73c2a75fbaa903ea50f135f6f99fbf24e07a3523/src/Microsoft.SqlTools.SqlCore/ObjectExplorer/SmoModel/DatabaseTreeNode.cs#L23
 I haven't investigated smo to see why that's happening but I am making this fix to unblock people on mwc

Comment (Charles-Gagnon) [reply src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs:44]: The number of SMO connection issues you're hitting is definitely concerning...

But regardless - so these failures would only be something that people hit in development? Since we won't have sql login connections once this goes live. Just want to make sure, although I guess this isn't really blocking. 

But please at least add a comment here explaining why it's only being opened when there's a token. (and ideally add context on why you're even opening it here in the first place too)

File: src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs [hunk 2]
```diff
@@ -127,7 +133,6 @@ public static async Task<TreeNode[]> ExpandTreeNode(TreeNode node, ObjectExplore
                                 if (!nodeContext.Server.ConnectionContext.IsOpen && securityToken != null)
                                 {
                                     var underlyingSqlConnection = nodeContext.Server.ConnectionContext.SqlConnectionObject;
-                                    underlyingSqlConnection.AccessToken = securityToken.Token;
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs:130]: Why is this being removed?

File: src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs [hunk 3]
```diff
@@ -38,10 +38,18 @@ public static async Task<TreeNode[]> Expand(string connectionString, SecurityTok
         {
             using (SqlConnection conn = new SqlConnection(connectionString))
             {
-                conn.AccessToken = accessToken?.Token;
-                conn.Open();
+                // In case of MWC access token based connections, we need to open the sql connection first before creating  the server connection
```
Comment (Charles-Gagnon) [inline src/Microsoft.SqlTools.SqlCore/ObjectExplorer/StatelessObjectExplorer.cs:41]: Is this actually related to MWC? I'd just remove that to avoid confusion for people who don't know what MWC is.

**Review by Charles-Gagnon — APPROVED** [review]
